src/api/db/migrate/20210603113812_fix_hostsystem_definition.rb

```diff
@@ -0,0 +1,17 @@
+class FixHostsystemDefinition < ActiveRecord::Migration[6.0]
+  class SchemaMigration < ApplicationRecord
+  end
+
+  def up
+    # no real production data inside yet, so we skip any conversation.
+    safety_assured { remove_column :repositories, :hostsystem_id }
```

Comment:
It doesn't matter if the column has production data or not, removing a column is **not** safe. Please follow the recommendations from [strong_migrations](https://github.com/ankane/strong_migrations#removing-a-column) on how to safely drop a column. Without this, we're going to be down once this migration runs.

Reply:
that way?

Reply:
1. Change the code to ignore the column you want to drop.
2. Deploy
3. Create a migration to remove the column
4. Deploy and run the migration.

This cannot happen all at the same time. It has to be done in two steps, so it will have to be in 2 PRs. It's a bit more work, but then we're not down because we remove a column.

Reply:
as I said, this column couldn't have used before, if it was the result would be broken anyway.

But we can move just the removal to another later migration if wanted.

Reply:
It doesn't matter if this column is used or not, ActiveRecord is caching the table columns. This is why it's crashing if you simply remove a column like you intended to.

Please remove the commented-out lines from the migration because this will be confusing anybody looking at the code.

Reply:
The database schema is also wrong now since the column `hostsystem_id` is still being removed.